python_modules/dagster/dagster/_core/definitions/asset_spec.py

```diff
@@ -148,6 +154,15 @@ def __new__(
         for owner in owners:
             validate_asset_owner(owner, key)
 
+        if kinds is not None and len(kinds) > 2:
+            raise DagsterInvalidDefinitionError("Assets can have at most two kinds currently.")
+
+        storage_kind = (tags or {}).get("dagster/storage_kind")
+        if kinds is not None and storage_kind is not None and storage_kind not in kinds:
+            raise DagsterInvalidDefinitionError(
+                f"If specifying dagster/storage_kind={storage_kind} and kinds, {storage_kind} must be in the list of kinds"
+            )
+
```

Comment:
I think we can eliminate this. No reason to support storage_kind at all in this new world